VstsTasks/AzureDtlCreateVM/task-funcs.ps1

```diff
@@ -414,9 +416,16 @@ function Wait-ApplyArtifacts
             {
                 throw "Waited for more than $(ConvertTo-MinutesString $totalWaitMinutes). Failing the task."
             }
+
+            $continueWaiting = $true
+
             $vm = Get-AzureRmResource -ResourceId $ResourceId
-            $provisioningState = $vm.Properties.provisioningState
-            $continueWaiting = $provisioningState -eq 'ApplyingArtifacts' -or $provisioningState -eq 'UpgradingVmAgent'
+            if ($vm)
+            {
+                $provisioningState = $vm.Properties.provisioningState
+                $continueWaiting = $provisioningState -and ($provisioningState -eq 'ApplyingArtifacts' -or $provisioningState -eq 'UpgradingVmAgent')

```

Comment:
didnt we also discuss that we will reverse this check. by checking for succeeded/failed state instead. all other states continue waiting

Reply:
We did, but I don't think it is required. I've added the actual state in the VSTS logs. So, we'll be able to tell if that additional logic is really required.